target/linux/mvebu/patches-6.6/700-mvneta-tx-queue-workaround.patch

```diff
@@ -0,0 +1,43 @@
+From: Felix Fietkau <nbd@nbd.name>
+Subject: mvneta: tx queue workaround
```

Comment:
Do we need still this hacky patch for kernel 6.6?

Reply:
reference [5411](https://github.com/openwrt/openwrt/issues/5411)

Reply:
That reference is in the patch description, but it does not answer my question. :-)

Reply:
Let's get some empirical data. I repeated my tests from 6.1 bump (https://github.com/openwrt/openwrt/pull/12938#issuecomment-1599133250).

**Test device:**
Linksys WRT1900ACS (mvebu/cortexa9)
OpenWrt SNAPSHOT, r0+25621-231d84e7a8 (f7c732bf9ed9bd09646a528681256fb7b0d80bfc with this PR on top)
Kernel 6.6.22
netperf 2.7.0-r3

**Test client:**
x86_64 Fedora 39 workstation
intel I219-V GBit Ethernet directly connected
netperf 2.7.1

**Test commands**
```
netperf -H 172.16.101.1 -l 60 -t TCP_MAERTS &
netperf -H 172.16.101.1 -l 60 -t TCP_MAERTS &
netperf -H 172.16.101.1 -l 60 -t TCP_MAERTS &
netperf -H 172.16.101.1 -l 60 -t TCP_MAERTS &
```

**Results with patch 700**

```
Recv   Send    Send                          
Socket Socket  Message  Elapsed              
Size   Size    Size     Time     Throughput  
bytes  bytes   bytes    secs.    10^6bits/sec
131072  16384  16384    60.00     234.15
131072  16384  16384    60.00     234.14
131072  16384  16384    60.00     234.14
131072  16384  16384    60.00     234.14
                                  936,57   (total)
```


**Results without patch 700**

```
Recv   Send    Send                          
Socket Socket  Message  Elapsed              
Size   Size    Size     Time     Throughput  
bytes  bytes   bytes    secs.    10^6bits/sec
131072  16384  16384    60.00     888.79
131072  16384  16384    60.01      15.95
131072  16384  16384    60.01      15.95
131072  16384  16384    60.01      15.92
                                  936,61   (total)
```

(both results repeatable with minor deviations)

**Conclusion**

Sadly the answer is "Yes", patch seems to be still necessary.

Reply:
Hmm, it would be interesting to see if it is even broken by using vanilla kernel, but I think that upstream Linux kernel developers would notice that.  Does anyone try to reach them? Isn't this caused for us by another hacky patch in this repository?

Reply:
Is software flow offload is working with 6.6.x?

Reply:
> Is software flow offload is working with 6.6.x?

From what I've gathered, it does not on devices that use the switch for both LAN and WAN, but it does for e.g. Turris Omnia which has a separate WAN connection. It is working on my Turris Omnia - with the usual quirks.